### DISCUSSION TITLE
What is the overhead of calling cookies many times.

### DISCUSSION THREAD

**Discussion by adr-x** [vercel/next.js#71327] [Help] [answered]
### Summary

I need to grab a jwt from cookies to make request to external API 
I'm trying to gauge the cost of calling cookies for each call to a different endpoint vs calling it once and passing it. 

```ts
// getX calls cookies() internally.
export default async function Page() {
 const  a = await getA()
 const  b = await getB()
 const  c = await getC()
}
```
vs 
```ts
import { cookies } from 'next/headers'
export default async function Page() {
const cookieStore = cookies()
const jwt = cookieStore.get('jwt')
 const  a = await getA(jwt)
 const  b = await getB(jwt)
 const  c = await getC(jwt)
}
```

**Comment by wilsonadenuga** [comment, marked as answer]
The second approach is generally more efficient. Calling cookies() once and reusing the JWT is better than calling it multiple times internally in each get function. This reduces overhead, especially if cookies() is an expensive operation.
However, you can further optimize this using Promise.all:

```javascript
import { cookies } from 'next/headers'

export default async function Page() {
  const cookieStore = cookies()
  const jwt = cookieStore.get('jwt')

  const [a, b, c] = await Promise.all([
    getA(jwt),
    getB(jwt),
    getC(jwt)
  ])
}

**Reply by adr-x** [reply]
thanks!, do you have insights into whether `cookies()` is expensive?

**Reply by wilsonadenuga** [reply]
Using cookies() isn't a major performance hit, but it's smart to optimize. Calling it once and reusing the result, like in your second example, is a good habit. The Promise.all() approach takes this further. For your average app, you won't see huge gains, but it's still worth doing. It keeps your code clean and efficient, which is always a win.
